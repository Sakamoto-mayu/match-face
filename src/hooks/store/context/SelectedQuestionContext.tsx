"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { selectedQuestionReducer } from "../reducer/selectedQuestionReducer";
import { SelectReducerAction } from "@/types/admin/tasks/register/types";

const SelectedQuestionContext = createContext<
  [string[], Dispatch<SelectReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], SelectReducerAction>>(
    selectedQuestionReducer,
    []
  );
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
