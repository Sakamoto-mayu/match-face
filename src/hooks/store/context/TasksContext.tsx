"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
