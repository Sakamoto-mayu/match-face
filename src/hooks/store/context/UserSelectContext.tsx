"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { userSelectReducer } from "../reducer/userSelectReducer";
import { UserSelectReducerAction } from "@/types/admin/tasks/register/types";

const UserSelectContext = createContext<
  [string[], Dispatch<UserSelectReducerAction>] | undefined
>(undefined);

export const CounterProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], UserSelectReducerAction>>(
    userSelectReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useCounter' must be used in Provider.");
  return context;
};
