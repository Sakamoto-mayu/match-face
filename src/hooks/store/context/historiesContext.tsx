'use client'
import { createContext, Dispatch, useContext, useReducer, Reducer } from "react";
import { Departments, SelectHistoryAction, Skills } from "@/types/admin/histories/admin_histories";
import { skills,departments } from "@/const/admin_histories";
import { SelectHistoryReducer } from "../reducer/historiesReducer";

const SelectHistoryContext = createContext<{month:string,department:string,skills:string[]}| undefined>(undefined);
const SelectHistoryDispatchContext = createContext<Dispatch<SelectHistoryAction>|undefined>(undefined)
const initialSelect = {month:"",department:"",skills:[]}

export function SelectHistoryProvider ({children}:{children:React.ReactNode}){
    const [state,dispatch] = useReducer(SelectHistoryReducer,initialSelect)
    return (
        <SelectHistoryContext.Provider value={state}>
            <SelectHistoryDispatchContext.Provider value={dispatch} >
            {children}
            </SelectHistoryDispatchContext.Provider>
        </SelectHistoryContext.Provider>
    )
}

export const useSelectHistory = () => {
    const context = useContext(SelectHistoryContext)
    if(context === undefined) {
        throw new Error("Function 'useSelectHistory' must be used in Provider.")
    }
    return context
}

export const useSelectHistoryDispatch = () => {
    const dispatch = useContext(SelectHistoryDispatchContext)
    if(dispatch === undefined) {
        throw new Error("Function 'useSelectHistoryDispatch' must be used in Provider.")
    }
    return dispatch
}

// skill,departmentのあまり変わらないデータは一旦contextからデータ取得
export const SkillsContext = createContext<Skills|undefined>(undefined)
export const DepartmentsContext = createContext<Departments|undefined>(undefined)

// constからのimport(valueの値)をfetchで取得したデータに変える
// const skills = fetch('')
// const departments = fetch('')

export function HistoriesProvider({children}:{children:React.ReactElement}) {
    return (
        <SkillsContext.Provider value = {skills}>
        <DepartmentsContext.Provider value = {departments}>
            {children}
        </DepartmentsContext.Provider>
        </SkillsContext.Provider>
    )
}
export const useSkills = () => {
    const context = useContext(SkillsContext)
    if(context === undefined){
        throw new Error('skillsが見つかりません')
    }
    return context
}
export const useDepartments = () => {
    const context = useContext(DepartmentsContext)
    if(context === undefined){
        throw new Error('departmentsが見つかりません')
    }
    return context
}
