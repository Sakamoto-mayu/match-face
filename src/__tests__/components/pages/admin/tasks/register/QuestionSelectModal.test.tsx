import QuestionSelectModal from "@/components/pages/admin/tasks/register/QuestionSelectModal ";
import { SelectedQuestionProvider } from "@/hooks/store/context/SelectedQuestionContext";
import { screen, render, fireEvent } from "@testing-library/react";

// Todo: 非同期関数実装後に自動テストの追加・修正の必要あり

// モーダル展開用関数
const open = () => {
  render(
    <SelectedQuestionProvider>
      <QuestionSelectModal />
    </SelectedQuestionProvider>
  );
  fireEvent.click(screen.getByRole("button"));
}

describe("QuestionSelectModal.tsx", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    open();
  })

  describe("スナップショットテスト", () => {
    it("モーダルを開いている状態", () => {
      const { baseElement } = render(
        <SelectedQuestionProvider>
          <QuestionSelectModal />
        </SelectedQuestionProvider>
      );
      expect(baseElement).toMatchSnapshot();
    });
  })

  describe("入力テスト", () => {
    it("検索ボックスの入力テスト", async () => {
      render(
        <SelectedQuestionProvider>
          <QuestionSelectModal />
        </SelectedQuestionProvider>
      );
      const input = screen.getByTestId("search-box") as HTMLInputElement;
      fireEvent.change(input, { target: { value: "案件" } });
      expect(input.value).toBe("案件");
    });
  })
})
