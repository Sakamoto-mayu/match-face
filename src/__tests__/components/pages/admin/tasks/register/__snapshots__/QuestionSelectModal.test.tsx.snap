// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionSelectModal.tsx スナップショットテスト モーダルを開いている状態 1`] = `
<body>
  <div />
  <div
    class="fixed top-0 left-0 z-30 flex items-center justify-center w-full h-full"
  >
    <div
      class="relative z-20 w-3/4 max-w-5xl py-7 px-10 bg-white"
    >
      <div
        class="flex flex-col items-center gap-8 w-full mx-auto"
      >
        <div
          class="mx-auto"
        >
          <div
            class="flex flex-col items-start gap-3"
          >
            <div
              class="text-base"
            >
              <h2>
                ▶️質問を選択する
              </h2>
            </div>
            <form
              class="flex flex-col item-center justify-center px-12 py-5 border-2 w-full"
            >
              <div
                class="flex flex-col items-center gap-5"
              >
                <div
                  class="flex flex-col items-center gap-5"
                >
                  <div
                    class="flex items-start justify-center gap-2 w-fit"
                  >
                    <input
                      class="border-light-gray border-2 text-xs p-2 lg:w-96 sm:w-80 w-72"
                      data-testid="search-box"
                      id="search"
                      type="text"
                      value=""
                    />
                    <button
                      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs ml-4 w-16"
                    >
                      検索
                    </button>
                  </div>
                  <div
                    class="flex items-start gap-4 flex-wrap"
                  >
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_1"
                        name="department"
                        type="checkbox"
                        value="Java"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_1"
                      >
                        Java
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_2"
                        name="department"
                        type="checkbox"
                        value="PHP"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_2"
                      >
                        PHP
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_3"
                        name="department"
                        type="checkbox"
                        value="FR"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_3"
                      >
                        FR
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_4"
                        name="department"
                        type="checkbox"
                        value="CL"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_4"
                      >
                        CL
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_5"
                        name="department"
                        type="checkbox"
                        value="ML"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_5"
                      >
                        ML
                      </label>
                    </div>
                    <div>
                      <input
                        class="peer hidden"
                        id="teck_6"
                        name="department"
                        type="checkbox"
                        value="QA"
                      />
                      <label
                        class="text-xs inline-flex items-center justify-center py-2 px-4 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                        for="teck_6"
                      >
                        QA
                      </label>
                    </div>
                  </div>
                  <div
                    class="flex mx-auto gap-4"
                  >
                    <span
                      class="text-xs"
                    >
                      使用技術：
                    </span>
                    <div
                      class="grid xl:grid-cols-7 lg:grid-cols-6 md:grid-cols-4 grid-cols-3 gap-3"
                    >
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_1"
                          name="skill"
                          type="checkbox"
                          value="JavaScript"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_1"
                        >
                          JavaScript
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_2"
                          name="skill"
                          type="checkbox"
                          value="TypeScript"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_2"
                        >
                          TypeScript
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_3"
                          name="skill"
                          type="checkbox"
                          value="React"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_3"
                        >
                          React
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_4"
                          name="skill"
                          type="checkbox"
                          value="Next.js"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_4"
                        >
                          Next.js
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_5"
                          name="skill"
                          type="checkbox"
                          value="Vue.js"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_5"
                        >
                          Vue.js
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_6"
                          name="skill"
                          type="checkbox"
                          value="Angular"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_6"
                        >
                          Angular
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_7"
                          name="skill"
                          type="checkbox"
                          value="Nuxt.js"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_7"
                        >
                          Nuxt.js
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_8"
                          name="skill"
                          type="checkbox"
                          value="jQuery"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_8"
                        >
                          jQuery
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_9"
                          name="skill"
                          type="checkbox"
                          value="Redux"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_9"
                        >
                          Redux
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_10"
                          name="skill"
                          type="checkbox"
                          value="vuex"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_10"
                        >
                          vuex
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_11"
                          name="skill"
                          type="checkbox"
                          value="MaterialUI"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_11"
                        >
                          MaterialUI
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_12"
                          name="skill"
                          type="checkbox"
                          value="tailwind"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_12"
                        >
                          tailwind
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_13"
                          name="skill"
                          type="checkbox"
                          value="Bootstrap"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_13"
                        >
                          Bootstrap
                        </label>
                      </div>
                      <div
                        class="flex"
                      >
                        <input
                          class="w-4 h-4 accent-orange cursor-pointer"
                          id="skill_14"
                          name="skill"
                          type="checkbox"
                          value="Vercel"
                        />
                        <label
                          class="text-xs ml-1"
                          for="skill_14"
                        >
                          Vercel
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
                <div>
                  <button
                    class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
                    type="submit"
                  >
                    絞り込み
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="w-full mx-auto"
        >
          <table
            class="border-collapse border-2 border-deep-gray text-sm mx-auto w-full"
          >
            <thead
              class="bg-light-gray"
            >
              <tr>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
                />
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-2/12"
                >
                  最終編集日
                </th>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
                >
                  案件名
                </th>
                <th
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  質問内容
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  <input
                    class="w-4 h-4 accent-orange cursor-pointer"
                    type="checkbox"
                    value="問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１"
                  />
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  2023-7-6
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  test1
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１
                </td>
              </tr>
              <tr>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  <input
                    class="w-4 h-4 accent-orange cursor-pointer"
                    type="checkbox"
                    value="問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２"
                  />
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  2023-7-6
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  test1
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２
                </td>
              </tr>
              <tr>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  <input
                    class="w-4 h-4 accent-orange cursor-pointer"
                    type="checkbox"
                    value="問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３"
                  />
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  2023-7-6
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  test1
                </td>
                <td
                  class="border-2 border-deep-gray px-4 py-2 text-center"
                >
                  問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
          >
            選択完了
          </button>
        </div>
      </div>
    </div>
    <div
      class="absolute top-0 left-0 w-full h-full bg-black opacity-30"
    />
  </div>
  <div>
    <button
      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
    >
      追加
    </button>
  </div>
</body>
`;
